src/dagmc/DagMC.cpp

```diff
@@ -212,6 +214,423 @@ ErrorCode DagMC::setup_indices() {
+    verts_to_delete.merge(vertices);
+  }
+
+  // get the union of the graveyard vertices' adjacencies.
+  // this should retrieve all elements connected to these vertices
+  // (edges, triangles, tets, etc. )
+  Range adj;
+  rval = moab_instance()->get_adjacencies(verts_to_delete, 1, true, adj,
+                                          Interface::UNION);
+  MB_CHK_SET_ERR(rval,
+                 "Failed to get dimension 1 adjacencies of graveyard vertices");
+  ents_to_delete.merge(adj);
+  adj.clear();
+  rval = moab_instance()->get_adjacencies(verts_to_delete, 2, true, adj,
+                                          Interface::UNION);
+  MB_CHK_SET_ERR(rval,
+                 "Failed to get dimension 2 adjacencies of graveyard vertices");
+  ents_to_delete.merge(adj);
+
+  // delete accumulated entity sets
+  rval = moab_instance()->delete_entities(sets_to_delete);
+  MB_CHK_SET_ERR(rval, "Failed to delete graveyard entity sets");
+
+  // delete accumulated entities
+  rval = moab_instance()->delete_entities(ents_to_delete);
+  MB_CHK_SET_ERR(rval, "Failed to delete graveyard entities");
+
+  // delete accumulated vertices (this must come last)
+  rval = moab_instance()->delete_entities(verts_to_delete);
+  MB_CHK_SET_ERR(rval, "Failed to delete graveyard vertices");
+
+  // re-construct the implicit complement's tree if needed
+  if (trees_exist && ic) {
+    rval = geom_tool()->construct_obb_tree(ic);
+    MB_CHK_SET_ERR(rval,
+                   "Failed to re-create the implicit complement OBBTree/BVH");
+  }
+
+  // update geometry sets in the GTT
+  rval = geom_tool()->find_geomsets();
+  MB_CHK_SET_ERR(rval,
+                 "Failed to find geometry sets after removing the graveyard");
+
+  // re-initialize DAGMC indices
+  rval = setup_indices();
+  MB_CHK_SET_ERR(rval, "Failed to setup DAGMC indices");
+
+  return MB_SUCCESS;
+}
+
+ErrorCode DagMC::create_graveyard(bool overwrite) {
+  /* Method summary
+    - Determine the global axis-aligned bounding box of the current model
+    - Create a new volume set for the graveyard
+    - Create a new group labeled as the graveyard and add volume to that group
+    - Create an inner surface from the bounding box
+    - Create an outer surface by extending the inner surface
+    - Add surfaces as children of the new graveyard volume and the implicit
+       complement
+    - Set the surface senses w.r.t. the new graveyard volume and implicit
+       complement
+    - Delete the implicit complement volume's out-of-date OBBTree/BVH
+    - Construct the new volume's OBBTree/BVH
+    - Construct the updated implicit complement OBBTree/BVH
+  */
+  ErrorCode rval;
+
+  // remove existing graveyard
+  if (overwrite) {
+    remove_graveyard();
+  }
+
+  // if a graveyard already exists and we aren't overwriting it,
+  // report an error
+  if (has_graveyard() && !overwrite) {
+    MB_CHK_SET_ERR(MB_FAILURE, "Graveyard already exists");
+  }
```

Comment:
maybe this can be combined together ?
```
if (overwrite) {}
else if(has_graveyard()) {}
```

Reply:
I don't follow... this error should only be raised in the case that a graveyard already exists and the overwrite flag is set to false. Separating the two would have a different meaning than the one that exists here. 

That being said, the check for `!overwrite` isn't really necessary, as the block above removes the graveyard if `overwrite` is `true`, meaning that at this point if a graveyard already exists we can raise an error.

Reply:
just line 392 you had a 
`if(overwrite){}`
and l 398
`if (has_greaveyard() && !overwrite){}`

my suggestion was to combine the two into a if() else if()

but your new implementation resolves it :)